### PR TITLE
Load CTLB using libbpf

### PR DESCRIPTION
**Pull request by tomastigera**
## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by sridhartigera** [comment]
Should we modify calculate-flags to remove __BPFTOOL_LOADER__ for ctlb files?

File: bpf/nat/connecttime.go [hunk 1]
```diff
@@ -84,50 +85,61 @@ func RemoveConnectTimeLoadBalancer(cgroupv2 string) error {
 	return nil
 }
```
Comment (sridhartigera) [inline bpf/nat/connecttime.go:86]: Should ```RemoveConnectTimeLoadBalancer``` as well be moved to ```libbpf``` APIs?

File: bpf/nat/connecttime.go [hunk 2]
```diff
@@ -84,50 +85,61 @@ func RemoveConnectTimeLoadBalancer(cgroupv2 string) error {
...
-	if err != nil {
-		err = errors.Wrapf(err, "failed to attach program %s", progName)
-		goto out
+	baseDir := "/sys/fs/bpf/tc/globals/"
+	for m, err := obj.FirstMap(); m != nil && err == nil; m, err = m.NextMap() {
+		// In case of global variables, libbpf creates an internal map <prog_name>.rodata
+		// The values are read only for the BPF programs, but can be set to a value from
+		// userspace before the program is loaded.
+		if m.IsMapInternal() {
+			log.WithField("prog", progName).Warn("Load-time configuration not supported, using defaults.")
```
Comment (fasaxc) [inline bpf/nat/connecttime.go:116]: When do we hit this?

Comment (tomastigera) [reply bpf/nat/connecttime.go:116]: When the program has globals. I am adding the globals in a follow up PR

File: bpf/nat/connecttime.go (hunk 1 above)
Comment (tomastigera) [reply bpf/nat/connecttime.go:86]: After some discussion, this is not straight forward and we will leave it out for now. It is only executed when felix restarts and we cannot remember links.

File: bpf/nat/connecttime.go (hunk 2 above)
Comment (tomastigera) [reply bpf/nat/connecttime.go:116]: This is the follow up https://github.com/projectcalico/felix/pull/3046

**Review by sridhartigera — APPROVED** [review]

**Pull request merged**